Error on compile

list.c: In function ‘set_rig_port’:
list.c:47:54: error: ‘FILPATHLEN’ undeclared (first use in this function)
   47 |   strncpy(rig->rig->state.rigport.pathname, portstr, FILPATHLEN - 1);
      |                                                      ^~~~~~~~~~
list.c:47:54: note: each undeclared identifier is reported only once for each function it appears in
make: *** [Makefile:42: list.o] Error 1

--- a/list.c
+++ b/list.c
@@ -44,7 +44,7 @@ struct rs_rig *add_rig(rig_model_t model)
 
 void set_rig_port(struct rs_rig *rig, const char *portstr)
 {
-  strncpy(rig->rig->state.rigport.pathname, portstr, FILPATHLEN - 1);
+  strncpy(rig->rig->state.rigport.pathname, portstr, HAMLIB_FILPATHLEN - 1);
 }
 
 void set_rig_speed(struct rs_rig *rig, int speed)

--- a/main.c
+++ b/main.c
@@ -81,8 +81,8 @@ int xmain(int argc, char **argv)
   rigb = rig_init(modelb);
 
   riga->state.rigport.parm.serial.rate = RIG_A_SPEED;
-  strncpy(riga->state.rigport.pathname, RIG_A_PORT, FILPATHLEN - 1);
-  strncpy(rigb->state.rigport.pathname, RIG_B_PORT, FILPATHLEN - 1);
+  strncpy(riga->state.rigport.pathname, RIG_A_PORT, HAMLIB_FILPATHLEN - 1);
+  strncpy(rigb->state.rigport.pathname, RIG_B_PORT, HAMLIB_FILPATHLEN - 1);
 
   retval = rig_open(riga);
   if(retval != RIG_OK) { fprintf(stderr, "*** Could not open FT857.\n"); exit(1); }

--- a/rigsync.h
+++ b/rigsync.h
@@ -26,6 +26,11 @@
 // note Ubuntu 20.04 uses hamlib 3.3
 #include <hamlib/rig.h>
 
+// Hamlib 4.x renames FILEPATHLEN to HAMLIB_FILEPATHLEN
+#ifndef HAMLIB_FILPATHLEN
+#define HAMLIB_FILPATHLEN FILPATHLEN
+#endif
+
 struct rs_rig_state
 {
   freq_t freq;

